group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -9421,33 +9421,44 @@ public void testClassicGroupMaybeDelete() {
 
     @Test
     public void testConsumerGroupDelete() {
+        String groupId = "group-id";
         GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
             .build();
-        context.groupMetadataManager.getOrMaybeCreateConsumerGroup("group-id", true);
+
+        // Create an empty group.
+        context.replay(RecordHelpers.newGroupEpochRecord(groupId, 10));
+        context.replay(RecordHelpers.newGroupSubscriptionMetadataRecord(groupId, Collections.emptyMap()));
+        context.replay(RecordHelpers.newTargetAssignmentEpochRecord(groupId, 10));
```

Comment:
You could actually replace this by using `withConsumerGroup` when constructing the context.